x-pack/filebeat/input/httpjson/config.go

```diff
@@ -33,18 +34,24 @@ type config struct {
 
 // Pagination contains information about httpjson pagination settings
 type Pagination struct {
-	IsEnabled        bool          `config:"enabled"`
+	Enabled          *bool         `config:"enabled"`
 	ExtraBodyContent common.MapStr `config:"extra_body_content"`
 	Header           *Header       `config:"header"`
 	IDField          string        `config:"id_field"`
 	RequestField     string        `config:"req_field"`
 	URL              string        `config:"url"`
 }
 
+// IsEnabled returns true if the `enable` field is set to true in the yaml.
+func (p *Pagination) IsEnabled() bool {
+	return p != nil && (p.Enabled == nil || *p.Enabled)
+}
+
 // HTTP Header information for pagination
 type Header struct {
-	FieldName    string `config:"field_name"`
-	RegexPattern string `config:"regex_pattern"`
+	FieldName    string `config:"field_name" validate:"required"`
+	RegexPattern string `config:"regex_pattern" validate:"required"`
```

Comment:
Does `Unpack` work when you use `RegexPattern *regexp.Regexp`? The documentation for go-ucfg suggests that it should work. Then you don't need to do it yourself in `Validate()`.

Reply:
I will check.